test/e2e/soak_k8petstore.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"syscall"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// readTransactions reads # of transactions from the k8petstore web server endpoint.
+// for more details see the source of the k8petstore web server.
+
+func readTransactions(c *unversioned.Client, ns string) (error, int) {
+	body, err := c.Get().
+		//TODO, need to fix this to work
+		//with node ports
+		Namespace(ns).
+		Prefix("proxy").
+		Resource("services").
+		Name("frontend").
+		Suffix("llen").
+		DoRaw()
+	if err != nil {
+		return err, -1
+	} else {
+		totalTrans, err := strconv.Atoi(string(body))
+		return err, totalTrans
+	}
+}
+
+// runK8petstore runs the k8petstore application, bound to external ip "ip", and polls it to assert that "min_expected"
+// transactions are acquired in a maximum of "max_seconds".
+func runK8petstore(ip string, restServers int, loadGenerators int, c *unversioned.Client, ns string, minExpected int, maxSeconds int64) {
+
+	var root0 = absOrDie(filepath.Clean(filepath.Join(path.Base(os.Args[0]), "..")))
```

Comment:
this is the path to the examples/ directory.   Renaming the var now so that its clear.